### DISCUSSION TITLE
How to use Sparkle in pyqt5

### DISCUSSION THREAD

**Discussion by duolabmeng6** [sparkle-project/Sparkle#2205] [Q&A]
I'm sorry to ask this question because I didn't find a solution when I read the document.

I use it. python PySide6 use pyinstaller Construction App 

Can I use sparkle to update my app automatically?

**Comment by duolabmeng6** [comment]
I saw this tutorial, but it didn't work.
https://fman.io/blog/codesigning-and-automatic-updates-for-pyqt-apps/

https://github.com/sparkle-project/Sparkle/issues/839

```
import sys

from PySide6.QtCore import QRect
from PySide6.QtWidgets import *


class MainWin(QMainWindow):
    def __init__(self):
        super().__init__()
        self.setGeometry(QRect(100, 100, 400, 300))
        self.setWindowTitle('Sparkle Test')
        # 创建按钮
        self.btn = QPushButton(self)
        self.btn.setText('Check for Updates')
        self.btn.setGeometry(QRect(20, 20, 200, 40))
        self.btn.show()
        # self.aboutToQuit.connect(about_to_quit)

        self.show()


QT_APP = QApplication([])

APPCAST_URL = 'http://127.0.0.1:8000/SampleAppcast.xml'
SPARKLE_PATH = '/Users/chensuilong/Downloads/Sparkle-2.2.0/Sparkle.framework'

from objc import pathForFramework, loadBundle

sparkle_path = pathForFramework(SPARKLE_PATH)
objc_namespace = dict()
loadBundle('Sparkle', objc_namespace, bundle_path=sparkle_path)


def about_to_quit():
    print('about to quit')
    # 见 https://github.com/sparkle-project/Sparkle/issues/839
    objc_namespace['NSApplication'].sharedApplication().terminate_(None)


QT_APP.aboutToQuit.connect(about_to_quit)
sparkle = objc_namespace['SUUpdater'].sharedUpdater()
sparkle.setAutomaticallyChecksForUpdates_(True)
sparkle.setAutomaticallyDownloadsUpdates_(True)
NSURL = objc_namespace['NSURL']
sparkle.setFeedURL_(NSURL.URLWithString_(APPCAST_URL))
sparkle.checkForUpdatesInBackground()

window = MainWin()
window.btn.clicked.connect(about_to_quit)

sys.exit(QT_APP.exec())

```

I downloaded version 2.2.0 from the official website.
Write the above script according to your tutorial
But it didn't work.

I hope there are bosses who have used pyqt to help.

**Comment by zorgiepoo** [comment]
You need to first figure out how to build an app bundle (with a Info.plist), bundle an Obj-C framework, and figure out the inter-opt that allows you to use such obj-c code. Once you figure that out, setting up Sparkle can be brought in the picture. Until then the problem is not that related to Sparkle.

**Reply by duolabmeng6** [reply]
Thank you very much for your answer.
I need to know something about Obj-C.

**Comment by duolabmeng6** [comment]
Thank you very much for your help

Since it can't be used and the principle of automatic update is not complicated, I built a wheel myself

I hope it can help people who need pyqt application to update automatically

https://github.com/duolabmeng6/qtAutoUpdateApp

Support MacOS system application update



Support window system application update



No server required



Use GitHub to automatically build and publish software



The client automatically checks the software resources published in GitHub and updates them automatically



Use the image to optimize the domestic network to avoid the failure to download updates

**Reply by Henxun** [reply]
does it need permission when extract file  to target?
